switchlink/switchlink_main.c

```diff
@@ -288,6 +289,13 @@ static void nl_process_event_loop(void) {
 
     ret = select(num_fds, &read_fds, NULL, NULL, NULL);
     if (ret == -1) {
+      // Select system call being interrupted by a signal EINTR.
+      // This is not really an error condition, hence continue the loop.
+      if (errno == EINTR)
+        continue;
+
+      krnlmon_log_error("Received select error, errno :%d, meaning: %s",
```

Comment:
Several observations:

1. `strerror` is not thread-safe (it works on an internal static buffer) and is deprecated. You should use a thread-safe variant such as `strerror_r` instead.
2. `errno` is also not thread-safe. It is suggested that you make a local copy and use that, especially if you're going to reference it more than once.
3. The logger is far more useful than the console in a production environment (which may well be headless). I would recommend dropping the call to `perror` if you're going log the error.
4. The formatting of the logger message is inconsistent. You have `errno :<number>` (label/space/colon/text) followed by `meaning: <text>` (label/colon/space/text). This may seem like a minor point, but it's less professional and it detracts from the customer experience.
5. The log message text doesn't say what the problem is. The `perror` output is even less useful. Both messages are for the benefit of the developer, not the customer. This is especially true of the decimal value of `errno`. I would suggest something along the lines of `"Error selecting Netlink event socket: <meaning>"`.